File: Level-1/69.java
import java.util.*;
class Solution {
    public int[] solution(String today, String[] terms, String[] privacies) {
        int[] answer;
        
        String[] to = today.split("\\.");
        int to_date = ((((Integer.parseInt(to[0]) * 12) + Integer.parseInt(to[1])) * 28) + Integer.parseInt(to[2]));
        
        HashMap<String, Integer> hash = new HashMap<>();
        for(String term : terms) {
            hash.put(term.split(" ")[0], Integer.parseInt(term.split(" ")[1]) * 28 - 1);
        }
        
        ArrayList<Integer> list = new ArrayList<>();
        for(int i = 0; i < privacies.length; i++) {
            int year = Integer.parseInt(privacies[i].split(" ")[0].split("\\.")[0]);
			int month = Integer.parseInt(privacies[i].split(" ")[0].split("\\.")[1]);
			int date = Integer.parseInt(privacies[i].split(" ")[0].split("\\.")[2]);
            
            int p_date = ((((year * 12) + month) * 28) + date) + hash.get(privacies[i].split(" ")[1]);
            
            if(p_date < to_date) {
                list.add(i + 1);
            }
        }
        
        answer = new int[list.size()];
        for(int i = 0; i < answer.length; i++) {
            answer[i] = list.get(i);
        }
        
        return answer;
    }
}

File: Level-1/69.js
function solution(today, terms, privacies) {
    const answer = [];
    const termType = {};
    
    const [cYear, cMonth, cDay] = today.split('.').map(v => +v);
    
    terms.forEach((item) => {
        const [type, term] = item.split(' ');
        termType[type] = +term;
    });
    
    privacies.forEach((item, index) => {
        const [date, type] = item.split(' ');
        const [year, month, day] = date.split('.').map(v => +v);
        
        let limitYear = year + Math.floor(termType[type] / 12),
            limitMonth = month + termType[type] % 12,
            limitDay = day - 1;
        
        if(limitDay === 0) {
            limitDay = 28;
            limitMonth -= 1;
        }
        
        if(limitMonth > 12) {
            limitMonth %= 12;
            limitYear += 1;
        }
        
        if(cYear > limitYear) {
            answer.push(index + 1);
        }else if(cYear === limitYear && cMonth > limitMonth) {
            answer.push(index + 1);
        }else if(cYear === limitYear && cMonth === limitMonth && cDay > limitDay) {
            answer.push(index + 1);
        }
    });
    
    return answer;
}
